source/simulator/parameters.cc

```diff
@@ -1124,6 +1146,124 @@ namespace aspect
+                   || (x_mapped_particle_properties.size() == 0),
+                   ExcMessage ("The list of names for the mapped particle "
+                               "property fields needs to either be empty or have a length equal to "
+                               "the number of compositional fields that are advected on particles."));
+
+      for (std::vector<std::string>::const_iterator p = x_mapped_particle_properties.begin();
+           p != x_mapped_particle_properties.end(); ++p)
+        {
+          // each entry has the format (white space is optional):
+          // <name> : <value (might have spaces)> [component]
+          //
+          // first tease apart the two halves
+          const std::vector<std::string> split_parts = Utilities::split_string_list (*p, ':');
+          AssertThrow (split_parts.size() == 2,
+                       ExcMessage ("The format for mapped particle properties  "
+                                   "requires that each entry has the form `"
+                                   "<name of field> : <particle property> [component]', "
+                                   "but there does not appear to be a colon in the entry <"
+                                   + *p
+                                   + ">."));
+
+          // the easy part: get the name of the compositional field
+          const std::string key = split_parts[0];
+
+          // check that the names used are actually names of fields,
+          // are solved by particles, and are unique in this list
+          std::vector<std::string>::iterator field_name_iterator = std::find(names_of_compositional_fields.begin(),
+                                                                             names_of_compositional_fields.end(), key);
+          AssertThrow (field_name_iterator
+                       != names_of_compositional_fields.end(),
+                       ExcMessage ("Name of field <" + key +
+                                   "> appears in the list of mapped particle properties, but"
+                                   "there is no field with this name."));
+
+          const unsigned int compositional_field_index = std::distance(names_of_compositional_fields.begin(),field_name_iterator);
+          AssertThrow (compositional_field_methods[compositional_field_index]
+                       == "particles",
+                       ExcMessage ("The field <" + key +
+                                   "> appears in the list of mapped particle properties, but"
+                                   "is not advected by a particle method."));
+
+          AssertThrow (std::count(names_of_compositional_fields.begin(),
+                                  names_of_compositional_fields.end(), key) == 1,
+                       ExcMessage ("Name of field <" + key +
+                                   "> appears more than once in the list of "
+                                   "mapped particle properties."));
```

Comment:
maybe name the parameter in which this happens to make it easier to find the place in the parameter file where this happens?